moto/iam/config.py

```diff
@@ -0,0 +1,173 @@
+import json
+
+from moto.core.exceptions import InvalidNextTokenException
+from moto.core.models import ConfigQueryModel
+from moto.iam import iam_backends
+
+
+class RoleConfigQuery(ConfigQueryModel):
+    def list_config_service_resources(
```

Comment:
Because the Config support is ... to say the least complex :), please clone over the docstrings similar to what I wrote for S3, but tailored for IAM.

This will be helpful for future implementations of resource types that are regional. Currently S3 is the only one implemented and S3 is a super edge case due to how it works (it's globally name-spaced but regionally resident).

Reply:
complex doesn't even begin to describe it. I have tried to add a bunch more comments